client/landing/stepper/declarative-flow/internals/steps-repository/link-in-bio-setup/index.tsx

```diff
@@ -89,31 +98,52 @@ const LinkInBioSetup: Step = function LinkInBioSetup( { navigation } ) {
 						onSelect={ setSelectedFile }
 						selectedFile={ selectedFile }
 					/>
-					<LinkInBioInput
-						inputId={ 'link-in-bio-input-name' }
-						inputValue={ siteTitle }
-						isValid={ false }
-						onChange={ onChange }
-						errorMessage={ __( 'Your site needs a name so your subscribers can identify you.' ) }
-						label={ __( 'Site name' ) }
-						disabled={ isLoading }
-					/>
-					<LinkInBioInput
-						inputId={ 'link-in-bio-input-description' }
-						inputValue={ tagline }
-						isValid={ true }
-						onChange={ onChange }
-						errorMessage={ __(
-							'Your site needs a brief description so your subscribers can identify you.'
+					<FormFieldset disabled={ isLoading }>
+						<FormLabel htmlFor="link-in-bio-input-name">{ __( 'Site name' ) }</FormLabel>
+						<FormInput
+							name="link-in-bio-input-name"
+							id="link-in-bio-input-name"
+							value={ siteTitle }
+							onChange={ onChange }
+							style={ {
+								backgroundImage: ! siteTitleError ? `url(${ greenCheckmarkImg })` : 'unset',
+								backgroundRepeat: 'no-repeat',
+								backgroundPosition: '95%',
+								paddingRight: ' 40px',
+							} }
```

Comment:
All this can be in CSS for both fields:

```scss
.link-in-bio-setup__site-title {
		&:not( [value=''] ) {
			background-image: url( 'calypso/assets/images/onboarding/green-checkmark.svg' );
		}
		background-repeat: no-repeat;
		background-position: 95%;
		padding-right: 40px;
	}
```

Reply:
Thank you for reviewing. I had a similar implementation before pushing... this implementation makes lot of sense, but it behaves weird on page load when the input is not empty. Please see the video attached, by having inline styling this issue does not happen. The green check mark style is being applied later and the button "animates" from left to the right because of the styling props.

![](https://d.pr/i/lUJ40X.jpg)